runtime/j9vm/jvm.c

```diff
@@ -3589,13 +3589,24 @@ JVM_LoadLibrary(const char* libName)
 	J9JavaVM* javaVM = (J9JavaVM*)BFUjavaVM;
 	J9InternalVMFunctions* vmFuncs = javaVM->internalVMFunctions;
 	J9VMThread* currentThread = vmFuncs->currentVMThread(javaVM);
+	UDATA flagVMAccess = 0;
 
-	Trc_SC_LoadLibrary_Entry(libName);
+	Trc_SC_LoadLibrary_Entry(currentThread, libName);
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+	flagVMAccess = currentThread->publicFlags & J9_PUBLIC_FLAGS_VM_ACCESS;
+	Trc_SC_LoadLibrary_FlagVMAccess(currentThread, flagVMAccess);
+	if (0 != flagVMAccess) {
```

Comment:
After entering the VM, you will certainly have VM access, so there's no need to test here.

Reply:
Removed the flag and check.